Swiftfin tvOS/Views/VideoPlayer/LiveTVPlayerViewController.swift

```diff
@@ -0,0 +1,900 @@
+		                                                userInfo: nil, repeats: false)
+	}
+
+	@objc
+	private func dismissTimerFired() {
+		hideOverlay()
+	}
+
+	private func stopOverlayDismissTimer() {
+		overlayDismissTimer?.invalidate()
+	}
+}
+
+// MARK: Confirm Close Overlay Timer
+
+extension LiveTVPlayerViewController {
+
+	private func restartConfirmCloseDismissTimer() {
+		self.confirmCloseOverlayDismissTimer?.invalidate()
+		self.confirmCloseOverlayDismissTimer = Timer.scheduledTimer(timeInterval: 5, target: self,
+		                                                            selector: #selector(confirmCloseTimerFired), userInfo: nil,
+		                                                            repeats: false)
+	}
+
+	@objc
+	private func confirmCloseTimerFired() {
+		hideConfirmCloseOverlay()
+	}
+
+	private func stopConfirmCloseDismissTimer() {
+		confirmCloseOverlayDismissTimer?.invalidate()
+	}
+}
+
+// MARK: VLCMediaPlayerDelegate
+
+extension LiveTVPlayerViewController: VLCMediaPlayerDelegate {
+
+	// MARK: mediaPlayerStateChanged
+
+	func mediaPlayerStateChanged(_ aNotification: Notification!) {
+
+		// Don't show buffering if paused, usually here while scrubbing
+		if vlcMediaPlayer.state == .buffering && viewModel.playerState == .paused {
+			return
+		}
+
+		viewModel.playerState = vlcMediaPlayer.state
+
+		if vlcMediaPlayer.state == VLCMediaPlayerState.ended {
+			if viewModel.autoplayEnabled && viewModel.nextItemVideoPlayerViewModel != nil {
+				didSelectPlayNextItem()
+			} else {
+				didSelectClose()
+			}
+		}
+	}
+
+	// MARK: mediaPlayerTimeChanged
+
+	func mediaPlayerTimeChanged(_ aNotification: Notification!) {
+
+		if !viewModel.sliderIsScrubbing {
+			viewModel.sliderPercentage = Double(vlcMediaPlayer.position)
+		}
+
+		// Have to manually set playing because VLCMediaPlayer doesn't
+		// properly set it itself
+		if abs(currentPlayerTicks - lastPlayerTicks) >= 10000 {
+			viewModel.playerState = VLCMediaPlayerState.playing
+		}
+
+		// If needing to fix subtitle streams during playback
+		if vlcMediaPlayer.currentVideoSubTitleIndex != viewModel.selectedSubtitleStreamIndex &&
+			viewModel.subtitlesEnabled
+		{
+			didSelectSubtitleStream(index: viewModel.selectedSubtitleStreamIndex)
+		}
+
+		if vlcMediaPlayer.currentAudioTrackIndex != viewModel.selectedAudioStreamIndex {
+			didSelectAudioStream(index: viewModel.selectedAudioStreamIndex)
+		}
+
+		lastPlayerTicks = currentPlayerTicks
+
+		// Send progress report every 5 seconds
+		if abs(lastProgressReportTicks - currentPlayerTicks) >= 500_000_000 {
+			viewModel.sendProgressReport()
+
+			lastProgressReportTicks = currentPlayerTicks
+		}
+	}
+}
+
+// MARK: PlayerOverlayDelegate
+
+extension LiveTVPlayerViewController: PlayerOverlayDelegate {
+
+	func didSelectAudioStream(index: Int) {
+		vlcMediaPlayer.currentAudioTrackIndex = Int32(index)
```

Comment:
(2) Through a variety of mechanisms, this is where the audio track gets set. From (1), the track index was -1 which is the index for "no audio track" in VLCKit. You might just have to remove this line.

Reply:
Bingo! Removing this made liveTV audio work for me. Thanks for the help.